kernel/kernel-api/src/main/java/io/delta/kernel/types/FieldMetadata.java

```diff
@@ -74,6 +76,63 @@ public Object get(String key) {
         return metadata.get(key);
     }
 
+    /**
+     * @param key  the key to check for
+     * @param type the type to cast the value to
+     * @return the value (casted to the given type) to which the specified key is mapped,
+     * or null if there is no mapping for the given key
+     */
+    private <T> T get(String key, Class<T> type) {
```

Comment:
Looks great. Can you please move all the `private` methods to be below the `public` methods?

This includes moving `private String valueToJson(Object value)`, too.

After that, LGTM!

Reply:
done